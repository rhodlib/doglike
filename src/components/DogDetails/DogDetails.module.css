.ActionButtons {
  padding-top: 2rem;
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

.Spinner {
  align-self: center;
}

.ActionButtons a {
  margin: 0 0.8rem;
}

.Image {
  align-self: center;
  background-color: rgba(255, 255, 255, 0.2);
  backdrop-filter: blur(0.25rem);
  box-shadow: 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 2px 2px 0 rgba(0, 0, 0, 0.14),
    0 1px 5px 0 rgba(0, 0, 0, 0.12);
  border-radius: 0.5rem;
  height: 100%;
  max-height: 60vh;
  max-width: 60vw;
  min-width: 20rem;
  object-fit: contain;
  width: 100%;
}

.DogDetails {
  display: flex;
  flex-direction: column;
}
