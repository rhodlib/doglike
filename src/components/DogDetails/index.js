import React from "react";
import PropTypes from "prop-types";
import styles from "./DogDetails.module.css";
import Loader from "../Loader";
import Title from "../Title";
import Back from "../Back/index";
import Reload from "../Reload/index";

/**
 * Component that receives a name and an image url and displays them.
 */
export const Card = ({ dogImg = "", dogName, onReload = () => undefined }) => (
  <div className={styles.DogDetails}>
    {dogImg === "" ? (
      <Loader className={styles.Spinner} />
    ) : (
      <>
        <Title name={dogName} />
        <img className={styles.Image} src={dogImg} alt={dogName} />
        <div className={styles.ActionButtons}>
          <Back to="/" />
          <Reload onClick={onReload} tabIndex={0} to={`/post/${dogName}`} />
        </div>
      </>
    )}
  </div>
);

Card.propTypes = {
  dogImg: PropTypes.string,
  dogName: PropTypes.string.isRequired
};

export default Card;
