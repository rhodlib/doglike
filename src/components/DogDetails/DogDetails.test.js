import React from "react";
import { render } from "@testing-library/react";
import DogDetails from "./index";
import { BrowserRouter as Router } from "react-router-dom";

it("Matches snapshot", () => {
  const { asFragment } = render(
    <Router>
      <DogDetails dogImg="test" dogName="test" />
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Without dogImage", () => {
  const { asFragment } = render(
    <Router>
      <DogDetails dogName="test" />
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});
