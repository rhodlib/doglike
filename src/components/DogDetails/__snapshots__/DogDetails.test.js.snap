// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="DogDetails"
  >
    <h1
      class="Title "
    >
      test
    </h1>
    <img
      alt="test"
      class="Image"
      src="test"
    />
    <div
      class="ActionButtons"
    >
      <a
        class="Back"
        href="/"
      />
      <a
        class="Reload"
        href="/post/test"
        tabindex="0"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Without dogImage 1`] = `
<DocumentFragment>
  <div
    class="DogDetails"
  >
    <div
      class="Loader Spinner"
    />
  </div>
</DocumentFragment>
`;
